### PR TITLE
Message::SetLength and Message::ResizeMessage Bug Fix

### PR DESCRIPTION
**Pull request by nibanks**
This updates Message::SetLength and Message::ResizeMessage to handle running out of buffers.

### CODE REVIEW

File: src/core/common/message.cpp [hunk 1]
```diff
@@ -148,6 +149,9 @@ ThreadError Message::ResizeMessage(uint16_t aLength)
     FreeBuffers(curBuffer);
 
     return kThreadError_None;
+
+exit:
```
Comment (jwhui) [inline src/core/common/message.cpp]: Code style: we prefer to have a single return point in a given function.  Just define a local variable for the error and set it in the `VerifyOrExit()`.

**Comment by jwhui** [comment]
Good catch!  Thanks.  Just a minor code-style comment inline.

File: .gitignore [hunk 2]
```diff
@@ -3,6 +3,7 @@
 *.gcno
```
Comment (jwhui) [inline .gitignore]: These additions are good.  However, we'd prefer to separate this out into a separate PR since they are unrelated to the bug being fixed.  Would also be good to add `output` to the list as well.

**Comment by jwhui** [comment]
LGTM 👍 

Please sign the CLA so that we can merge this PR.

**Comment by nibanks** [comment]
Unfortunately, I am still awaiting approval on my side. I will sign when that goes through.

**Comment by nibanks** [comment]
I signed it!

File: src/core/common/message.cpp [hunk 3]
```diff
@@ -134,6 +136,7 @@ ThreadError Message::ResizeMessage(uint16_t aLength)
         if (curBuffer->GetNextBuffer() == NULL)
         {
             curBuffer->SetNextBuffer(NewBuffer());
+            VerifyOrExit(curBuffer->GetNextBuffer() != NULL, error = kThreadError_NoBufs;);
```
Comment (jwhui) [inline src/core/common/message.cpp]: nit: the `;` after `kThreadError_NoBufs` is not necessary.

**Comment by jwhui** [comment]
LGTM 👍
